syntaxes/dart.json

```diff
@@ -180,67 +180,63 @@
 		"constants-and-special-vars": {
 			"patterns": [
 				{
-					"match": "\\b(true|false|null)\\b",
+					"match": "(?<!\\$)\\b(true|false|null)\\b(?!\\$)",
 					"name": "constant.language.dart"
 				},
 				{
-					"match": "\\b(this|super)\\b",
+					"match": "(?<!\\$)\\b(this|super)\\b(?!\\$)",
 					"name": "variable.language.dart"
 				},
 				{
-					"match": "\\b((0(x|X)[0-9a-fA-F]*)|(([0-9]+\\.?[0-9]*)|(\\.[0-9]+))((e|E)(\\+|-)?[0-9]+)?)\\b",
+					"match": "(?<!\\$)\\b((0(x|X)[0-9a-fA-F]*)|(([0-9]+\\.?[0-9]*)|(\\.[0-9]+))((e|E)(\\+|-)?[0-9]+)?)\\b(?!\\$)",
 					"name": "constant.numeric.dart"
 				},
 				{
-					"match": "\\b_?[A-Z][a-zA-Z0-9_]+\\b",
+					"match": "(?<![a-zA-Z0-9_$])[_$]*[A-Z][a-zA-Z0-9_$]*",
 					"name": "support.class.dart"
 				},
 				{
-					"match": "(_?[a-z][a-zA-Z0-9_]+)(\\(|\\s+=>)",
+					"match": "([_$]*[a-z][a-zA-Z0-9_$]*)(\\(|\\s+=>)",
 					"captures": {
 						"1": {
 							"name": "entity.name.function.dart"
 						}
 					}
-				},
-				{
-					"match": "\\b[A-Z_0-9]+\\b",
```

Comment:
Sorry, I should've made this explicit! This regex intended to detect and highlight constants as a separate sort of entity, but it's wrong for two reasons:

1. It's subsumed by the regex for classes, so it would actually only cause single capital letters to be highlighted differently (_although, as you noticed, depending on your color scheme selection in VSCode this may not have any effect at all_).

2. `SCREAMING_CAPS` isn't an appropriate way to look for constants in Dart anyway, seeing as [Effective Dart](https://www.dartlang.org/guides/language/effective-dart/style#prefer-using-lowercamelcase-for-constant-names) explicitly recommends `lowerCamelCase`.